sql/src/main/java/io/crate/execution/dml/upsert/CheckConstraints.java

```diff
@@ -56,6 +61,13 @@
             inputs.add(ctx.add(notNullRef));
         }
         expressions = ctx.expressions();
+        checkConstraints = table
+            .checkConstraints()
+            .stream()
+            .collect(Collectors.toMap(
+                CheckConstraint::name,
+                chk -> new Tuple<>(ctx.add(chk.cargo()), chk.expressionStr())
+            ));
```

Comment:
I think we could use a simple `ArrayList` here instead of a map? The `name` should be unique after all.

The area here is performance sensitive, so we should avoid expensive and heavy structures if possible.